From 7e5fd9cec22424616f9c929f6edc07845a46d322 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Wed, 7 Jun 2023 10:47:48 +0200
Subject: [PATCH 0402/1141] arm64: dts: st: add GPU on stm32mp25

Add GPU support for stm32mp25

Signed-off-by: Pierre-Yves MORDRET <pierre-yves.mordret@foss.st.com>
Change-Id: Ib901afe556d318f40cd2e7c8064fc1bbcf7206a7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/312377
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -29,6 +29,18 @@
 				clocks = <&rcc CK_BUS_VENC>;
 				feature-domains = <&rifsc STM32MP25_RIFSC_VENC_ID>;
 			};
+
+			gpu: gpu@48280000 {
+				compatible = "vivante,gc";
+				reg = <0x48280000 0x800>;
+				interrupts = <GIC_SPI 183 IRQ_TYPE_LEVEL_HIGH>;
+				resets = <&rcc GPU_R>;
+				clock-names = "axi", "core";
+				clocks = <&rcc CK_BUS_GPU>, <&rcc CK_KER_GPU>;
+				gpu-supply =  <&scmi_vddgpu>;
+				feature-domains = <&rifsc STM32MP25_RIFSC_GPU_ID>;
+				status = "disabled";
+			};
 		};
 	};
 };
